crud/select/comparators.lua

```diff
@@ -83,6 +83,33 @@ local function gen_array_cmp_func(target, key_parts)
     end
 end
 
+local function update_key_parts_by_field_names(space_format, field_names, key_parts)
+    if field_names == nil then
+        return key_parts
+    end
+
+    local fields_positions = {}
+    local updated_key_parts = {}
+    local last_position = #field_names + 1
+
+    for i, field_name in ipairs(field_names) do
+        fields_positions[field_name] = i
+    end
+
+    for _, part in ipairs(key_parts) do
+        local field_name = space_format[part.fieldno].name
+        if not fields_positions[field_name] then
+            fields_positions[field_name] = last_position
+            last_position = last_position + 1
+        end
+        local updated_part = table.copy(part)
+        updated_part.fieldno = fields_positions[field_name]
+        table.insert(updated_key_parts, updated_part)
```

Comment:
Could you just make something like `table.insert(updated_key_parts, {field_name = ..., fieldno = ..., is_nullable = ...})`?
"part" is not so huge and complex table.